crates/ast/src/visit.rs

```diff
@@ -4,139 +4,154 @@ use crate::ast::*;
 use solar_data_structures::trustme;
 use solar_interface::{Ident, Span};
 use solar_macros::declare_visitors;
+use std::ops::ControlFlow;
 
 declare_visitors! {
     /// AST traversal.
     pub trait Visit VisitMut <'ast> {
-        fn visit_source_unit(&mut self, source_unit: &#mut SourceUnit<'ast>) {
+        fn visit_source_unit(&mut self, source_unit: &#mut SourceUnit<'ast>) -> ControlFlow<()> {
```

Comment:
We should let implementors define the `Break` type; it should be possible to add `type BreakValue;` in the trait and these return types to be `ControlFlow<Self::BreakValue>`.

One can specify `Never` as the `BreakValue`, this should be in the docs for the type